### ISSUE TITLE
FE?/BE Using PATCH, PUT, DELETE instead of POST on every update or delete event

### ISSUE THREAD

**Issue by mertssmnoglu** [bluewave-labs/bluewave-uptime#560]
**Bluewave Uptime** provides a nice API but it's not completely restful. I know that there is no restriction or specific rule for making all routes restful but using the correct keywords/methods make it easy to read and manage the endpoints.

> ![restful-api-text-on-the-bluewave-labs-tech-stack-page](https://github.com/user-attachments/assets/e88449d3-e22b-43b3-a4c7-ca3fb9015a6a)
> https://bluewavelabs.ca/our-tech-stack/

I'm started to write about this topic because of the documentation in README but when I open the source code some of the correct methods are actually provided by #533 but I think some some updates are still needed. We can also update the documentation as well. (I can open a new issue about it if needed)

TL;DR

Current(According to docs) | Suggestion | Provided by #533
|-------|-------|------|
`POST` /monitors/delete/{monitorId} | `DELETE` /monitor/{id} | :heavy_check_mark: 
`POST` /checks/delete/{monitorId} | `DELETE` /checks/{monitorId} | :x: 
`POST` /monitors/edit/{monitorId} | `PATCH OR PUT` /monitor/{monitorId} | -
...|...|...

- [x] I want to contribute on providing correct http request methods

**Comment by ajhollid** [comment]
Hi @mertssmnoglu,

Quite right indeed!  We've been meaning to go through and update the HTTP verbs, as you noticed we did some in #533.

We'd love to have your help in sorting out the rest of them if you're willing and able!  We would need to update

1.  Backend routes
2.  FE requests that use those routes
3.  API documentation in the readme.

I'll assign the issue to you, drop a message here if you need any clarification on anything :+1: 

Thanks for your interest in the project and looking forward to working together!
